planetiler-core/src/test/java/com/onthegomap/planetiler/pmtiles/PmtilesTest.java

```diff
@@ -96,44 +113,178 @@ void testRoundtripHeader() {
 
   @Test
   void testBadHeader() {
-    assertThrows(FileFormatException.class, () -> Pmtiles.Header.fromBytes(new byte[0]));
-    assertThrows(FileFormatException.class, () -> Pmtiles.Header.fromBytes(new byte[127]));
+    assertThrows(FileFormatException.class, () -> WriteablePmtiles.Header.fromBytes(new byte[0]));
+    assertThrows(FileFormatException.class, () -> WriteablePmtiles.Header.fromBytes(new byte[127]));
   }
 
   @Test
   void testRoundtripDirectoryMinimal() {
-    ArrayList<Pmtiles.Entry> in = new ArrayList<>();
-    in.add(new Pmtiles.Entry(0, 0, 1, 1));
+    ArrayList<WriteablePmtiles.Entry> in = new ArrayList<>();
+    in.add(new WriteablePmtiles.Entry(0, 0, 1, 1));
 
-    List<Pmtiles.Entry> out = Pmtiles.deserializeDirectory(Pmtiles.serializeDirectory(in));
+    List<WriteablePmtiles.Entry> out = WriteablePmtiles.deserializeDirectory(WriteablePmtiles.serializeDirectory(in));
     assertEquals(in, out);
   }
 
   @Test
   void testRoundtripDirectorySimple() {
-    ArrayList<Pmtiles.Entry> in = new ArrayList<>();
+    ArrayList<WriteablePmtiles.Entry> in = new ArrayList<>();
 
     // make sure there are cases of contiguous entries and non-contiguous entries.
-    in.add(new Pmtiles.Entry(0, 0, 1, 0));
-    in.add(new Pmtiles.Entry(1, 1, 1, 1));
-    in.add(new Pmtiles.Entry(2, 3, 1, 1));
+    in.add(new WriteablePmtiles.Entry(0, 0, 1, 0));
+    in.add(new WriteablePmtiles.Entry(1, 1, 1, 1));
+    in.add(new WriteablePmtiles.Entry(2, 3, 1, 1));
 
-    List<Pmtiles.Entry> out = Pmtiles.deserializeDirectory(Pmtiles.serializeDirectory(in));
+    List<WriteablePmtiles.Entry> out = WriteablePmtiles.deserializeDirectory(WriteablePmtiles.serializeDirectory(in));
     assertEquals(in, out);
-    out = Pmtiles.deserializeDirectory(Pmtiles.serializeDirectory(in, 0, in.size()));
+    out = WriteablePmtiles.deserializeDirectory(WriteablePmtiles.serializeDirectory(in, 0, in.size()));
     assertEquals(in, out);
   }
 
   @Test
   void testRoundtripDirectorySlice() {
-    ArrayList<Pmtiles.Entry> in = new ArrayList<>();
+    ArrayList<WriteablePmtiles.Entry> in = new ArrayList<>();
 
     // make sure there are cases of contiguous entries and non-contiguous entries.
-    in.add(new Pmtiles.Entry(0, 0, 1, 0));
-    in.add(new Pmtiles.Entry(1, 1, 1, 1));
-    in.add(new Pmtiles.Entry(2, 3, 1, 1));
+    in.add(new WriteablePmtiles.Entry(0, 0, 1, 0));
+    in.add(new WriteablePmtiles.Entry(1, 1, 1, 1));
+    in.add(new WriteablePmtiles.Entry(2, 3, 1, 1));
 
-    List<Pmtiles.Entry> out = Pmtiles.deserializeDirectory(Pmtiles.serializeDirectory(in, 1, 2));
+    List<WriteablePmtiles.Entry> out = WriteablePmtiles.deserializeDirectory(
+      WriteablePmtiles.serializeDirectory(in, 1, 2));
     assertEquals(1, out.size());
   }
+
+  @Test
+  void testBuildDirectoriesRootOnly() throws IOException {
+
+  }
+
+  @Test
+  void testBuildDirectoriesLeaves() throws IOException {
+
+  }
```

Comment:
empty tests?

Reply:
Intending to add some more edge cases here around the adaptive directory sizing